### PR TITLE
docs: refine the pull request template

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve? 

This PR refines the pull request template, guide the PR author to write better the pull request description.

### What is changed and how it works?
1. put `What problem does this PR solve` on the first section.
This is the section the reviewers most care about.

2. add `how it works` in the end of `What is changed`.
The auther tend to answer the title question and ignore the long description in the comments, include `how it works` reminds the author to explain the `How`.

3. Make a checklist to fill out.
Makes the YES/NO questions easier to answer, simplify the templates and added more questions.

### Check List

Tests 
 - No code

### CODE REVIEW

**Comment by coocood** [comment]
@zz-jason @shenli PTAL

**Comment by zz-jason** [comment]
The check list makes PR description confusion.

**Comment by coocood** [comment]
@shenli @zz-jason PTAL

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -2,73 +2,39 @@
...
 
-## How has this PR been tested? (mandatory)
+### Check List
 
-<!--
-Please describe the tests that you ran to verify your changes.
-Have you finished unit tests, integration tests, or manual tests?
-What additional tests would give you greater confidence in this change?
--->
+<!-- Please REMOVE the items that don't apply. -->
```
Comment (CaitinChen) [inline .github/pull_request_template.md]: Please REMOVE the items that **are not applicable**.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -2,73 +2,39 @@
...
--->
+Code changes
 
+ - Has exported function/method change
+ - Has exported variable/fields change
+ - Has interface methods change
+ - Has persistent data change
 
-## Refer to a related PR or issue link (optional)
+Side effect
```
Comment (CaitinChen) [inline .github/pull_request_template.md]: Side effects

**Comment by CaitinChen** [comment]
LGTM

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -2,73 +2,39 @@
 Thank you for contributing to TiDB! Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
 -->
 
-## What have you changed? (mandatory)
+### What problem does this PR solve? <!--add issue link with summary if exists-->
```
Comment (zz-jason) [inline .github/pull_request_template.md]: how about: "add issue link with summary" and remove "if exists" to force every PR related to an issue.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -2,73 +2,39 @@
...
 
-## How has this PR been tested? (mandatory)
+### Check List
 
-<!--
-Please describe the tests that you ran to verify your changes.
-Have you finished unit tests, integration tests, or manual tests?
-What additional tests would give you greater confidence in this change?
--->
+<!-- Please REMOVE the items that are not applicable. -->
```
Comment (zz-jason) [inline .github/pull_request_template.md]: how about:
```
### Check List <!-- REMOVE the items that are not applicable. -->
```

**Comment by shenli** [comment]
LGTM

**Comment by zz-jason** [comment]
LGTM

**Pull request merged**